libs/common/src/lib/media/media.service.ts

```diff
@@ -1,10 +1,39 @@
 import { Injectable } from "@nestjs/common";
 import { ConfigService } from "@nestjs/config";
 import { Media } from "@terramatch-microservices/database/entities";
+import { PutObjectCommand, S3Client } from "@aws-sdk/client-s3";
 
 @Injectable()
 export class MediaService {
-  constructor(private readonly configService: ConfigService) {}
+  private readonly s3: S3Client;
+
+  constructor(private readonly configService: ConfigService) {
+    this.s3 = new S3Client({
+      region: this.configService.get<string>("AWS_REGION"),
+      credentials: {
+        accessKeyId: this.configService.get<string>("AWS_ACCESS_KEY_ID") ?? "",
+        secretAccessKey: this.configService.get<string>("AWS_SECRET_ACCESS_KEY") ?? ""
+      }
+    });
+  }
+
+  async uploadFile(
+    buffer: any,
+    key: string,
+    contentType: string,
+    bucket: string = this.configService.get<string>("AWS_BUCKET") ?? ""
+  ) {
+    const command = new PutObjectCommand({
+      Bucket: bucket,
+      Key: key,
+      Body: buffer,
+      ContentType: contentType,
+      ACL: "public-read"
+    });
+
+    await this.s3.send(command);
+    console.log(`Uploaded ${key}`);
```

Comment:
We'll want to make use of TMLogger instead of console.log in the final code. This one seems like a reasonable thing to promote to a real log.